ranger/config.go

```diff
@@ -0,0 +1,50 @@
+package ranger
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/xy-planning-network/trails"
+	"github.com/xy-planning-network/trails/http/middleware"
+	"github.com/xy-planning-network/trails/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+type Config[User RangerUser] struct {
```

Comment:
Nitpick: for some reason, my mind gets misdirected by the type name of `User`. I prefer the style of a single letter `config[U RangerUser]` and the cascading changes.